subsys/bluetooth/host/hci_core.c

```diff
@@ -391,6 +391,12 @@ int bt_hci_cmd_send(uint16_t opcode, struct net_buf *buf)
 {
 	struct bt_hci_cmd_hdr *hdr;
 
+	/* Make sure the HCI transport hasn't been closed before attempting anything else */
+	if (atomic_test_bit(bt_dev.flags, BT_DEV_DISABLE)) {
+		net_buf_unref(buf);
+		return -EHOSTDOWN;
+	}
+
```

Comment:
Do you need something similar for ACL and ISO Data being sent?

Reply:
Are those public APIs? I did this for `bt_hci_cmd_send()` (which also covers `bt_hci_cmd_send_sync()`) since those are public APIs. Anyway, IMO we can incrementally plug any other holes in the public API if those exist, i.e. not absolutely necessary in this PR. I did notice that we *do* have many public APIs that check for `BT_DEV_READY`, however it's possible that the coverage is not 100% complete.